src/UI.cpp

```diff
@@ -1729,6 +1728,7 @@ unsigned long getHexValue(const char sTitle[], unsigned long defValue) {
                     LCD.print_P(3, 11, UIStrings::Generic::LESS_SYM);
                 }
                 else {
+                    multiplier = ulpow(16, (digits - cursorPos - 1));
```

Comment:
Looking at the code here, the cause of the bug isn't immediately obvious; under what conditions does the bug occur?

Aside from the bug itself, where you moved the multiplier calculation to is a better place for it to be calculated anyway, as it would avoid recomputing the multiplier when it isn't needed.